src/ol/control/Control.js

```diff
@@ -124,14 +124,14 @@ class Control extends BaseObject {
     if (this.map_) {
       const target = this.target_
         ? this.target_
-        : map.getOverlayContainerStopEvent();
+        : this.map_.getOverlayContainerStopEvent();
```

Comment:
This change and the others here that change `map` to `this.map_` is not necessary, because of the assignment in line 123.

Reply:
Did you mean to add `| null` in line 112?

Reply:
Indeed, this is the same behaviour but TS can't figure that out. `map` will not be narrowed down by the if check where `this.map` will be. 
I did this change indeed because I wanted to add null here, but I reverted that part since I wasn't sure since it already supported undefined in this function.